--- a/support/truncate_repositories.sh	2013-08-26 11:59:22.000000000 +0200
+++ b/support/truncate_repositories.sh	2013-09-10 21:31:30.135841009 +0200
@@ -1,12 +1,12 @@
 #!/bin/bash
 
-home_dir="/home/git"
+repos_path="$(ruby -ryaml -e 'puts YAML::load_file("/etc/gitlab-shell.yml")["repos_path"]')"
 
 echo "Danger!!! Data Loss"
 while true; do
-  read -p "Do you wish to delete all directories (except gitolite-admin.git) from $home_dir/repositories/ (y/n) ?:  " yn
+  read -p "Do you wish to delete all directories (except gitolite-admin.git) from $repos_path (y/n) ?:  " yn
   case $yn in
-    [Yy]* ) sh -c "find $home_dir/repositories/. -maxdepth 1  -not -name 'gitolite-admin.git' -not -name '.' | xargs rm -rf"; break;;
+    [Yy]* ) sh -c "find $repos_path/. -maxdepth 1  -not -name 'gitolite-admin.git' -not -name '.' | xargs rm -rf"; break;;
     [Nn]* ) exit;;
     * ) echo "Please answer yes or no.";;
   esac
