
diff -ru lightdm-gtk-greeter-1.8.5.orig/src/lightdm-gtk-greeter.c lightdm-gtk-greeter-1.8.5/src/lightdm-gtk-greeter.c
--- lightdm-gtk-greeter-1.8.5.orig/src/lightdm-gtk-greeter.c	2014-05-10 18:56:28.100394660 +0200
+++ lightdm-gtk-greeter-1.8.5/src/lightdm-gtk-greeter.c	2014-05-10 19:39:20.007116506 +0200
@@ -2513,7 +2513,10 @@
     gtk_accel_map_add_entry ("<Login>/a11y/font", GDK_KEY_F1, 0);
     gtk_accel_map_add_entry ("<Login>/a11y/contrast", GDK_KEY_F2, 0);
     gtk_accel_map_add_entry ("<Login>/a11y/keyboard", GDK_KEY_F3, 0);
-    gtk_accel_map_add_entry ("<Login>/power/shutdown", GDK_KEY_F4, GDK_MOD1_MASK);
+    gtk_accel_map_add_entry ("<Login>/power/hibernate", GDK_KEY_H, GDK_CONTROL_MASK);
+    gtk_accel_map_add_entry ("<Login>/power/restart", GDK_KEY_R, GDK_CONTROL_MASK);
+    gtk_accel_map_add_entry ("<Login>/power/shutdown", GDK_KEY_S, GDK_CONTROL_MASK);
+    gtk_accel_map_add_entry ("<Login>/power/suspend", GDK_KEY_S, GDK_SHIFT_MASK | GDK_CONTROL_MASK);
 
 #ifdef START_INDICATOR_SERVICES
     init_indicators (config, &indicator_pid, &spi_pid);
diff -ru lightdm-gtk-greeter-1.8.5.orig/src/lightdm-gtk-greeter.glade lightdm-gtk-greeter-1.8.5/src/lightdm-gtk-greeter.glade
--- lightdm-gtk-greeter-1.8.5.orig/src/lightdm-gtk-greeter.glade	2014-05-10 18:56:28.100394660 +0200
+++ lightdm-gtk-greeter-1.8.5/src/lightdm-gtk-greeter.glade	2014-05-10 19:38:15.065114428 +0200
@@ -62,6 +62,7 @@
                       <object class="GtkMenuItem" id="suspend_menuitem">
                         <property name="visible">True</property>
                         <property name="can_focus">False</property>
+                        <property name="accel_path">&lt;Login&gt;/power/suspend</property>
                         <property name="label" translatable="yes">Suspend</property>
                         <signal name="activate" handler="suspend_cb" swapped="no"/>
                       </object>
@@ -70,6 +71,7 @@
                       <object class="GtkMenuItem" id="hibernate_menuitem">
                         <property name="visible">True</property>
                         <property name="can_focus">False</property>
+                        <property name="accel_path">&lt;Login&gt;/power/hibernate</property>
                         <property name="label" translatable="yes">Hibernate</property>
                         <signal name="activate" handler="hibernate_cb" swapped="no"/>
                       </object>
@@ -78,6 +80,7 @@
                       <object class="GtkMenuItem" id="restart_menuitem">
                         <property name="visible">True</property>
                         <property name="can_focus">False</property>
+                        <property name="accel_path">&lt;Login&gt;/power/restart</property>
                         <property name="label" translatable="yes">Restart...</property>
                         <signal name="activate" handler="restart_cb" swapped="no"/>
                       </object>
