--- a/Makefile
+++ b/Makefile
@@ -28,7 +28,7 @@
 all: 		atop atopsar
 
 atop:		atop.o    $(ALLMODS) Makefile
-		$(CC) atop.o $(ALLMODS) -o atop -lncurses -lz -lm -lrt $(LDFLAGS)
+		$(CC) $(CFLAGS) $(LDFLAGS) atop.o $(ALLMODS) -o atop $(shell ${PKG_CONFIG} --libs ncurses) -lm -lz -lrt
 
 atopsar:	atop
 		ln -sf atop atopsar
@@ -49,6 +49,8 @@
			then mkdir -p $(DESTDIR)$(LOGPATH); fi
		if [ ! -d $(DESTDIR)$(BINPATH) ]; 	\
			then mkdir -p $(DESTDIR)$(BINPATH); fi
+		if [ ! -d $(DESTDIR)$(SBINPATH) ]; 	\
+			then mkdir -p $(DESTDIR)$(SBINPATH); fi
		if [ ! -d $(DESTDIR)$(SCRPATH) ]; 	\
			then mkdir -p $(DESTDIR)$(SCRPATH); fi
		if [ ! -d $(DESTDIR)$(MAN1PATH) ]; 	\

